packages/block-editor/src/components/rich-text/use-paste-handler.js

```diff
@@ -257,3 +260,16 @@ function removeWindowsFragments( html ) {
 
 	return html.replace( startReg, '' ).replace( endReg, '' );
 }
+
+/**
+ * Removes the charset meta tag inserted by Chromium.
+ * See:
+ * - https://github.com/WordPress/gutenberg/issues/33585
+ * - https://bugs.chromium.org/p/chromium/issues/detail?id=1264616#c4
+ *
+ * @param {string} html the html to be stripped of the meta tag.
+ * @return {string} the cleaned html
+ */
+function removeCharsetMetaTag( html ) {
+	return html.replace( /^<meta charset=["']utf-8["']>/, '' );
```

Comment:
Since that we have the exact [source](https://source.chromium.org/chromium/chromium/src/+/main:ui/base/clipboard/clipboard_mac.mm;l=397-408) in chromium, I don't think we have to do regex search here. We can simply do:

```js
const metaTag = '<meta charset='utf-8'>';
if (html.startsWith(metaTag)) {
  return html.slice(metaTag.length);
}
```

and maybe it's a little bit faster.

Reply:
Will that match on both single and double quotes around the `charset` attribute?

Reply:
Ahh I made a typo in the comment, I updated it.

> Will that match on both single and double quotes around the charset attribute?

I don't think we have to worry about double quotes, no? Since we can clearly see that it's being hard-coded as single quotes in the source.